apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPoolOptions.cs

```diff
@@ -0,0 +1,233 @@
+        /// is applied to calls to create, refresh and delete sessions, as well as beginning transactions.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This value cannot be negative, but it can be 0. The behavior of a 0 value depends on <see cref="SpannerSettings.AllowImmediateTimeouts"/>.
+        /// The default value is 60.
+        /// </para>
+        /// </remarks>
+        public int Timeout
+        {
+            get => _timeout;
+            set => _timeout = GaxPreconditions.CheckNonNegative(value, nameof(value));
+        }
+
+        /// <summary>
+        /// The maximum number of session create operations allowed to occur simultaneously per session pool.
+        /// Spanner has limits on the number of sessions that can be created concurrently without affecting performance.
+        /// This value is not typically changed.
+        /// </summary>
+        /// <para>
+        /// This value must be positive. The default value is 10.
+        /// </para>
+        public int MaximumConcurrentSessionCreates
+        {
+            get => _maximumConcurrentSessionCreates;
+            set => _maximumConcurrentSessionCreates = GaxPreconditions.CheckArgumentRange(value, nameof(value), 1, int.MaxValue);
+        }
+
+        /// <summary>
+        /// The delay between maintenance loop iterations. Defaults to 30 seconds.
+        /// </summary>
+        internal TimeSpan MaintenanceLoopDelay
+        {
+            get => _maintenanceLoopDelay;
+            set => _maintenanceLoopDelay = CheckPositiveTimeSpan(value);
+        }
+
+        /// <summary>
+        /// Jitter to apply to session refresh times. Defaults to a 10% proportionally random jitter.
+        /// </summary>
+        internal RetrySettings.IJitter SessionRefreshJitter
+        {
+            get => _sessionRefreshJitter;
+            set => _sessionRefreshJitter = GaxPreconditions.CheckNotNull(value, nameof(value));
+        }
+
+        /// <summary>
+        /// Jitter to apply to session eviction times. Defaults to a 10% proportionally random jitter.
+        /// </summary>
+        internal RetrySettings.IJitter SessionEvictionJitter
+        {
+            get => _sessionEvictionJitter;
+            set => _sessionEvictionJitter = GaxPreconditions.CheckNotNull(value, nameof(value));
+        }
+
+        /// <summary>
+        /// Clock used for timings; can be replaced for testing.
+        /// </summary>
+        internal IClock Clock
+        {
+            get => _clock;
+            set => _clock = GaxPreconditions.CheckNotNull(value, nameof(value));
+        }
+
+        /// <summary>
+        /// Clock used for timings; can be replaced for testing.
+        /// </summary>
+        internal IScheduler Scheduler
+        {
+            get => _scheduler;
+            set => _scheduler = GaxPreconditions.CheckNotNull(value, nameof(value));
+        }
+
+        private static TimeSpan CheckPositiveTimeSpan(TimeSpan value)
```

Comment:
TODO: Move to GaxPreconditions?

Reply:
Will add the TODO, but I think we'd only do it if we see other uses.